### PR TITLE
feat: add simple push/pop transition on ios

### PR DESCRIPTION
**Pull request by WoLewicki**
## Description

Added a simple transition on iOS, similar to the default one, but without the animation of header and shadow. The code has been mainly taken from https://stackoverflow.com/a/45827454/13006595. Requested in #628.

## Changes

Added new `stackAnimation`: `simple_push`. Added proper handling on iOS and Android.

## Test code and steps to reproduce

`Test761.tsx` in `TestsExample` project.

## Checklist

- [x] v4 implementation
- [x] `rtl` on swipe detection
- [x] Included code example that can be used to test this change
- [x] Updated TS types
- [x] Updated documentation: 
  - [x] https://github.com/software-mansion/react-native-screens/blob/master/README.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/master/native-stack/README.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/master/createNativeStackNavigator/README.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/master/src/types.tsx
  - [x] https://github.com/software-mansion/react-native-screens/blob/master/src/native-stack/types.tsx
- [x] Ensured that CI passes

### CODE REVIEW

**Comment by stasundr** [comment]
@WoLewicki Thank you for this one! Any chances it'll be merged soon?

File: createNativeStackNavigator/README.md [hunk 1]
```diff
@@ -145,6 +145,7 @@ How the given screen should appear/disappear when pushed or popped at the top of
 - `default` - Uses a platform default animation.
 - `fade` - Fades screen in or out.
 - `flip` – Flips the screen, requires stackPresentation: `modal` (iOS only).
+- `"simple_push"` – performs a default animation, but without shadow and native header transition (iOS only)
```
Comment (WoLewicki) [inline createNativeStackNavigator/README.md:148]:
```suggestion
- `simple_push` – performs a default animation, but without shadow and native header transition (iOS only)
```

**Comment by WoLewicki** [comment]
I added first look at having the custom transition also on swipe and also implemented an event called during every frame or swipe position change with the current transition progress.

**Comment by tomtargosz** [comment]
Looks awesome! Would love to see this in a release 😄

**Pull request merged**